### DISCUSSION TITLE
How to get LSP to work?

### DISCUSSION THREAD

**Discussion by Yuxin0610** [NvChad/NvChad#1514] [Q&A] [answered]
I am new to nvChad and Lsp server, so this may be a dumb question to ask. 

Can anyone tell me how to get Lsp server to work with NvChad? :pray::pray:  (I am using it for C).

I try to place my cursor on a function and do `:LspDefinition`, but it didn't jump to the function definition.

Instead this is what I get:
![Screen Shot 2022-08-26 at 17 38 08](https://user-images.githubusercontent.com/30561368/187007332-77e0453e-8be6-40f0-8e96-ae88695ed884.png)

This is what my NvChad config file looks like:
![Screen Shot 2022-08-26 at 17 38 42](https://user-images.githubusercontent.com/30561368/187007333-3652a009-c318-4f21-93a2-6fee3f6a8966.png)
![Screen Shot 2022-08-26 at 17 38 47](https://user-images.githubusercontent.com/30561368/187007337-fb77b4e8-e164-4528-9473-e58ed4561d7d.png)

I have tried both `:MasonInstallAll` and `:PackerCompile`.

Is there something that I am doing wrong here?

**Comment by siduck** [comment, marked as answer]
please read the lsp docs in the site :((((((((((((((((((((

**Reply by Yuxin0610** [reply]
I think I have done everything listed in http://nvchad.com/config/Lsp%20stuff. Is there something I am missing?

**Reply by siduck** [reply]
did u do packercompile?

**Reply by Yuxin0610** [reply]
Yeah, I tried that but it didn't work.

**Reply by siduck** [reply]
idk why it shouldnt work other than not reading docs properly! Even made a video regarding setting it up from scratch 

https://www.youtube.com/watch?v=B-bK-FJ--H8

**Reply by Yuxin0610** [reply]
Just tried the pyright lsp on a python project, and it worked, so I think there is something wrong with my C package instead of LSP. Thank you for your timely response.

**Reply by siduck** [reply]
install clang

**Reply by mahmoudajawad** [reply]
> I think I have done everything listed in http://nvchad.com/config/Lsp%20stuff. Is there something I am missing?

Is this link, @siduck still supposed to be present in docs?

**Reply by siduck** [reply]
@mahmoudajawad oops that page's name was updated

https://nvchad.com/config/Language%20Server%20Protocol

**Reply by vkolgi** [reply]
Updating the link to the doc.
https://nvchad.com/docs/config/lsp
